R/check_suggested.R

```diff
@@ -39,22 +36,36 @@ check_suggested <- function(packages,
     return(FALSE)
   }
 
-  # If TRUE, fail if package is not installed
-  for (pkg in packages) {
-    if (!is_installed(pkg)) {
-      if (is_testing()) {
-        if (is.null(message)) {
-          message <- glue("`{top_level_fun}()` needs the '{pkg}' package.")
-        }
+  # Return early if all packages are installed.
+  if (rlang::is_installed(packages, version = version)) {
```

Comment:
> This early exit will work regardless of the value of use ?

Yes.


> I don't like the version argument in this function, in particular since version specifications can be passed to the pkg argument of check_installed() . Should we do an early installed <- map_lgl(...) and take it from there?

`version` must be a vector of the same length as `packages`. will error otherwise. This is how `rlang::check_installed()` works.


If we have a loop then, it would have to be `version[i]` indeed.

Reply:
Let's remove the `version` argument here.

``` r
rlang::check_installed("dplyr (>= 2.0.0)")
#> Error:
#> ! The package "dplyr" (>= 2.0.0) is required.
```

<sup>Created on 2023-10-16 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
I did this.